src/main/java/seedu/address/model/event/fields/Recurrence.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.model.event.fields;
+
+import java.util.Locale;
+import java.util.Objects;
+
+import seedu.address.model.event.enums.Interval;
+
+/**
+ * Represents an Events's Recurrence in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidInterval(String)}
+ */
+public class Recurrence {
+
+    public static final String MESSAGE_CONSTRAINTS = "Interval must be either: "
+            + "Daily, Weekly, Monthly or None";
+
+    public final Interval interval;
+
+    /**
+     * Constructs a {@code Recurrence}.
+     *
+     * @param interval A valid name.
+     */
+    public Recurrence(String interval) {
+        switch (interval.toLowerCase(Locale.ROOT)) {
+
+        case "none":
+            this.interval = Interval.NONE;
+            break;
```

Comment:
Magic string could be replaced; for the subsequent `case` statements also.

```suggestion
        case Interval.NONE.getValue():
            this.interval = Interval.NONE;
            break;
```